src/mono/wasm/runtime/src/mono/debug.ts

```diff
@@ -159,8 +159,7 @@ export function mono_wasm_call_function_on(request: CallRequest) {
     const fn_args = request.arguments != undefined ? request.arguments.map(a => JSON.stringify(a.value)) : [];
     const fn_eval_str = `var fn = ${request.functionDeclaration}; fn.call (proxy, ...[${fn_args}]);`;
 
-    const local_eval = eval; // https://rollupjs.org/guide/en/#avoiding-eval
-    const fn_res = local_eval(fn_eval_str);
+    const fn_res = eval(fn_eval_str);
```

Comment:
You could perhaps do something like this:

```ts
const fn_eval_str = `function (proxy) { var fn = ${request.functionDeclaration}; fn.call (proxy, ...[${fn_args}]); }`;
const fn_defn = local_eval (fn_eval_str);
const fn_res = fn_defn (proxy);
```

Reply:
Or maybe we should just use the `Function` constructor.

```ts
const fn_defn = new Function ('proxy', `var fn = ${request.functionDeclaration}; fn.call (proxy, ...[${fn_args}]);`);
const fn_res = fn_defn (proxy);
```

Reply:
Sorry to code golf this.  I'm confused why we need `eval` at all.

Why isn't it just:
```ts
  const fn = request.functionDeclaration;
  const fn_res = fn.call (proxy, ...fn_args);
```

Reply:
>   const fn = request.functionDeclaration;

`request.functionDeclaration` will be a string (eg. https://github.com/dotnet/runtime/blob/main/src/mono/wasm/debugger/DebuggerTestSuite/CallFunctionOnTests.cs#L26 and https://github.com/dotnet/runtime/blob/main/src/mono/wasm/debugger/DebuggerTestSuite/CallFunctionOnTests.cs#L79). Would this work? If so, then we could do what you are suggesting.

Reply:
It doesn't compile, I'm testing other approach based on @lambdageek suggestions.

Reply:
> > const fn = request.functionDeclaration;
> 
> `request.functionDeclaration` will be a string (eg. https://github.com/dotnet/runtime/blob/main/src/mono/wasm/debugger/DebuggerTestSuite/CallFunctionOnTests.cs#L26 and https://github.com/dotnet/runtime/blob/main/src/mono/wasm/debugger/DebuggerTestSuite/CallFunctionOnTests.cs#L79). Would this work? If so, then we could do what you are suggesting.

Ah ok.  So we really do need to eval it.

I think the `new Function(...)` approach should work.

Reply:
Sorry everybody, I was not aware that debugger tests are not running on CI :(